storage/postgres_store.go

```diff
@@ -0,0 +1,124 @@
+package storage
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/arhsxro/platform-go-challenge/config"
+	"github.com/arhsxro/platform-go-challenge/models"
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq"
+)
+
+type PostgresStore struct {
+	db *sqlx.DB
+}
+
+func isValidAssetType(paramType string) bool {
+	for _, validType := range models.ValidAssetTypes {
+		if models.AssetType(paramType) == validType {
+			return true
+		}
+	}
+	return false
+}
+
+func NewPostgresStore(cfg *config.Config) (*PostgresStore, error) {
+	db, err := NewPostgresDB(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to database: %v", err)
+	}
+
+	return &PostgresStore{db: db}, nil
+}
+
+// Retrieves a user's favorite assets from the database
+func (store *PostgresStore) GetUserFavorites(ctx context.Context, userID, filterType string, page, pageSize int) ([]models.Asset, error) {
+
+	var assets []models.Asset
+	var query string
+	var err error
+	offset := (page - 1) * pageSize
+	if filterType != "" {
+		if !isValidAssetType(filterType) {
+			log.Println("Invalid asset type")
+			return nil, errors.New("invalid asset type")
+		}
+		query = "SELECT asset_id, type, description, data FROM assets WHERE user_id = $1 and type = $2 LIMIT $3 OFFSET $4"
+		err = store.db.SelectContext(ctx, &assets, query, userID, filterType, pageSize, offset)
+	} else {
+		query = "SELECT asset_id, type, description, data FROM assets WHERE user_id = $1 LIMIT $2 OFFSET $3"
+		err = store.db.SelectContext(ctx, &assets, query, userID, pageSize, offset)
+	}
+
+	return assets, err
+}
+
+// Adds a new favorite asset for a user in the database
+func (store *PostgresStore) AddFavorite(ctx context.Context, userID string, asset models.Asset) error {
+	tx, err := store.db.Begin() //begin transaction
```

Comment:
Could you describe why you chose to use transactions?

Reply:
The truth is that I wasn't 100% sure if i should use a transaction since it only adds/delete/edit  a single asset to the db.Allthough i thought that maybe i get at the same time 2 requests where the first 1 is to delete an asset and the second one is to edit the same asset.So in this case with the use of transaction they do not interfere with each other, maintaining the integrity of the data.Another scenario could be that 1 request is to get the favorite assets and the 2 request to add an asset.